core/src/main/java/org/jdbi/v3/Foreman.java

```diff
@@ -34,8 +37,15 @@
         this.factories.addAll(factories);
     }
 
-    Argument waffle(Class<?> expectedType, Object it, StatementContext ctx)
+    Argument waffle(Type expectedType, Object it, StatementContext ctx)
     {
+        return waffle(new TypeResolver().resolve(expectedType), it, ctx);
+    }
+
+    Argument waffle(ResolvedType expectedType, Object it, StatementContext ctx)
+    {
+        ResolvedType objectType = new TypeResolver().resolve(Object.class);
```

Comment:
Oh, I see your comment in the broader context. I just extracted this to a var for reuse in two different blocks, on lines 60 and 65 below.